(*s3File).Open panics opening file with Content-Encoding: gzip

### Steps to reproduce

1. Create a file in S3, "s3://the-bucket/the-path".
2. Set `Content-Encoding: gzip` metadata on the file.
3. Call `(*s3File).Open` on that file. See the small example program below.

### Actual result
Get a panic that looks like:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8c5c05]

goroutine 6 [running]:
github.com/grailbio/base/file/s3file.newInfo(...)
	/home/ubuntu/code/base/file/s3file/s3file.go:726
github.com/grailbio/base/file/s3file.(*s3File).handleStat(0xc0000bd900, 0xbb7240, 0xc0000240e0, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0000927e0)
	/home/ubuntu/code/base/file/s3file/s3file.go:761 +0x425
github.com/grailbio/base/file/s3file.(*s3File).handleRequests(0xc0000bd900)
	/home/ubuntu/code/base/file/s3file/s3file.go:657 +0x1a7
created by github.com/grailbio/base/file/s3file.(*s3Impl).internalOpen
	/home/ubuntu/code/base/file/s3file/s3file.go:457 +0x1f2
exit status 2
```

### Expected result
No panic and correct file information.


### Example program
```go
package main

import (
	"context"

	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/grailbio/base/file"
	"github.com/grailbio/base/file/s3file"
)

func init() {
	file.RegisterImplementation("s3", func() file.Implementation {
		return s3file.NewImplementation(
			s3file.NewDefaultProvider(session.Options{}),
			s3file.Options{},
		)
	})
}

func main() {
	const path = "s3://path/to/file/with/content-encoding/gzip"
	_, err := file.Open(context.Background(), path)
	if err != nil {
		panic(err)
	}
}
```

--- a/file/s3file/s3file.go
+++ b/file/s3file/s3file.go
@@ -739,26 +739,27 @@ func (f *s3File) handleStat(req request) {
 	policy := newRetryPolicy(clients, f.opts)
 	for {
 		var ids s3RequestIDs
-		output, err := policy.client().GetObjectWithContext(ctx, &s3.GetObjectInput{
+		output, err := policy.client().HeadObjectWithContext(ctx, &s3.HeadObjectInput{
 			Bucket: aws.String(f.bucket),
 			Key:    aws.String(f.key)},
 			captureRequestIDs(&ids))
-		if policy.shouldRetry(req.ctx, err, f.name) {
+		if policy.shouldRetry(ctx, err, f.name) {
 			continue
 		}
 		if err != nil {
 			req.ch <- response{err: annotate(err, ids, &policy, "s3file.stat", f.name)}
 			return
 		}
-		if output.Body == nil {
-			panic(ids.String() + ": GetObject with nil Body")
-		}
-		output.Body.Close() // nolint: errcheck
 		if *output.ETag == "" {
 			req.ch <- response{err: errors.E("read", f.name, errors.NotExist, "awsrequestID:", ids.String())}
 			return
 		}
-		f.info = newInfo(f.name, output)
+		f.info = &s3Info{
+			name:    filepath.Base(f.name),
+			size:    *output.ContentLength,
+			modTime: *output.LastModified,
+			etag:    *output.ETag,
+		}
 		req.ch <- response{err: nil}
 		return
 	}

